docs/comparison.md

```diff
@@ -0,0 +1,139 @@
+# Comparison of `DataFrames` with R and Pandas
+
+R and Pandas (with Python) are commonly used data analysis tools. Both allow defining dataframes and provide a range of convenient ways of using them. Here, we translate some of the essential functionality of both languages. This is a good place to start if you know how to work with dataframes in R or Pandas and quickly want to get going using Julia.
+
+If you are instead interested in a general comparison of the languages, you can find that in the [base Julia documentation](http://docs.julialang.org/en/release-0.3/manual/noteworthy-differences/).
+
+For the examples below, we are following the [Pandas documentation](http://pandas.pydata.org/pandas-docs/stable/comparison_with_r.html) as much as possible. Since we use random numbers throughout, we assume that the ``Distributions`` package is loaded in Julia.
+
+# R
+
+## Indexing
+
+In R, columns of a `data.frame` can be accessed by name:
+
+```R
+df <- data.frame(a=rnorm(5), b=rnorm(5), c=rnorm(5), d=rnorm(5), e=rnorm(5))
+df[, c("a", "c")]
+
+df <- data.frame(matrix(rnorm(1000), ncol=100))
+df[, c(1:10, 25:30, 40, 50:100)]
+```
+
+Using DataFrames, the equivalent would be the following:
+
+```julia
+d = Uniform()
+df = DataFrame(a=rand(d, 5), b=rand(d, 5), c=rand(d, 5), d=rand(d, 5), e=rand(d, 5))
+df[[:a, :c]]
+
+d = Normal()
+data = rand(d, (1000, 100))
+df = convert(DataFrame, data)
+df[:, [1:10, 25:30, 40, 50:100]]
+```
+
+The biggest difference is that columns are not accessed using symbols like `:a` instead of strings like `"a"`.
+
+Further, note that `DataFrames` does not support named rows.
+
+## Aggregating data using `aggregate`
+
+```R
+df <- data.frame(
+  v1 = c(1,3,5,7,8,3,5,NA,4,5,7,9),
+  v2 = c(11,33,55,77,88,33,55,NA,44,55,77,99),
+  by1 = c("red", "blue", 1, 2, NA, "big", 1, 2, "red", 1, NA, 12),
+  by2 = c("wet", "dry", 99, 95, NA, "damp", 95, 99, "red", 99, NA, NA))
+aggregate(x=df[, c("v1", "v2")], by=list(mydf2$by1, mydf2$by2), FUN = mean)
+```
+
+In Julia, you would use ``aggregate``
+
+```julia
+df = DataFrame(
+  v1 = @data([1,3,5,7,8,3,5,NA,4,5,7,9]),
+  v2 = @data([11,33,55,77,88,33,55,NA,44,55,77,99]),
+  by1 = @data(["red", "blue", 1, 2, NA, "big", 1, 2, "red", 1, NA, 12]),
+  by2 = @data(["wet", "dry", 99, 95, NA, "damp", 95, 99, "red", 99, NA, NA]))
+
+aggregate(df, [:by1, :by2], mean)
+```
+
+Note as well that `DataFrames` requires the use of the `@data` macro to create the columns to ensure that missing values are correctly handled.
```

Comment:
I'd clarify that `@data` is required only to wrap vector literals containing `NA`s.